src/lib/handle-result.ts

```diff
@@ -26,11 +35,15 @@ export function handleResult(entry: EntryVariants, functionMap: FunctionMap, sec
         case Variant.EndOfRecord:
             // we've left the test module, to prevent name conflicts clear the current FunctionMap
             functionMap.clear()
-
+            functionLeaders.clear()
             return true
+        case Variant.FunctionLeader:
+            functionLeaders.set(entry.index, entry)
```

Comment:
Please move this insert operation to the `createUpdateFunctionSummary` function so that it can either insert or update the `Map` value if the expected order doesn't match the given one.